### PR TITLE
feat: ledger initialization stuff in django admin

### PR DESCRIPTION
**Pull request by iloveagent57**
**Description:**
* Some small developer QOL stuff.
* Better local development instructions in README.
* Remove docs from quality checks and `ci.yml`.
* Reasonable first pass at allowing for weak/strong admin editing ability depending on environment settings.
* Simple, first attempt at an idempotency key utility methods for ledgers and transactions that optionally take a subsidy and initial deposit, resp.
* Allow blank idp keys on the `Ledger` model, and set to a sane default if not provided on `save()`.
* Remove `JPY` as an allowed unit.
* `api.create_ledger()` now seeds the ledger with an optional initial deposit.
* Check if we're already inside a transaction when setting `durable=True` in `create_transaction()`.

Part of https://2u-internal.atlassian.net/browse/ENT-6894

**Reviewers:**
- [ ] tag reviewer 
- [ ] tag reviewer 

**Merge checklist:**
- [ ] All reviewers approved
- [x] CI build is green
- [x] Version bumped
- [x] Changelog record added
- [x] Documentation updated (not only docstrings)
- [x] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

File: openedx_ledger/api.py [hunk 1]
```diff
@@ -63,19 +67,37 @@ def reverse_full_transaction(transaction, idempotency_key, **metadata):
...
+            ledger,
+            initial_deposit,
+            is_initial_deposit=True,
+        )
+        create_transaction(
+            ledger,
+            initial_deposit,
+            initial_idpk,
+            state=models.TransactionStateChoices.COMMITTED,
+        )
```
Comment (pwnage101) [inline openedx_ledger/api.py:101]: You may want to take a look at a pattern that I've come up with on the subsidy side of things:

https://github.com/openedx/enterprise-subsidy/blob/f6500ac9b50a30319081aa7530b2b2f1eab42575/enterprise_subsidy/apps/subsidy/models.py#L191-L203

I'm open to either method.  One thing that would have to be done in order to shift the logic into this area of the code is a moderate amount of un-tangling the unit tests and test factories to use this different assumption about how the initial transaction gets created.

Comment (iloveagent57) [reply openedx_ledger/api.py:101]: Yeah I just started diving into that this afternoon.

**Comment by iloveagent57** [comment]
I'm not sure why we keep hitting this codecov issue:
```
[2023-03-16T13:26:30.111Z] ['info'] Pinging Codecov: https://codecov.io/upload/v4?package=github-action-3.1.1-uploader-0.3.5&token=*******&branch=aed%2Fledger-create-interface&build=4437553429&build_url=https%3A%2F%2Fgithub.com%2Fopenedx%2Fopenedx-ledger%2Factions%2Fruns%2F4437553429&commit=8728b5a9d98dc985b511b4507a80fa0c92daf06f&job=Python+CI&pr=10&service=github-actions&slug=openedx%2Fopenedx-ledger&name=&tag=&flags=unittests&parent=
[2023-03-16T13:26:30.641Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 500 - {"error": "Server Error (500)"}
Error: Codecov: Failed to properly upload: The process '/home/runner/work/_actions/codecov/codecov-action/v3/dist/codecov' failed with exit code 255
```
but the tests are passing, can probably just be merged when ready

**Review by pwnage101 — APPROVED** [review]
LGTM!

**Comment by pwnage101** [comment]
This just needs a rebase to grab a one-line change I merged.

**Pull request merged**